### PR TITLE
112/add from address field

### PR DESCRIPTION
**Pull request by friedger**
Add a from address field
If the user changes the address this is uses as the (global) current address

screenshot_1516315429

### CODE REVIEW

**Comment by ligi** [comment]
looks good - but does not compile for me - did you forget to add a file?

```
/home/kontinuum/kontinuum/workspace/walleth/walleth/app/src/main/res/layout/activity_create_transaction.xml:33:42-57: AAPT: No resource found that matches the given name (at 'layout_toLeftOf' with value '@id/scan_button').
    
/home/kontinuum/kontinuum/workspace/walleth/walleth/app/build/intermediates/res/merged/withGethWithFirebaseForPlay/debug/layout/activity_create_transaction.xml:30: error: Error: No resource found that matches the given name (at 'layout_toLeftOf' with value '@id/scan_button').

```

File: app/src/main/java/org/walleth/activities/CreateTransactionActivity.kt [hunk 1]
```diff
@@ -54,6 +54,9 @@ import java.math.BigInteger.ONE
 import java.math.BigInteger.ZERO
 import java.text.ParseException
 
+val TO_ADDRESS_REQUEST_CODE = 1
```
Comment (ligi) [inline app/src/main/java/org/walleth/activities/CreateTransactionActivity.kt]: should be const

**Comment by friedger** [comment]
issues addressed :-) 
for the layout issue, I am using a newer android plugin (and not running the tests :-p
for the const, thanks for teaching me kotlin, I am still learning.

File: app/src/main/java/org/walleth/activities/CreateTransactionActivity.kt [hunk 2]
```diff
@@ -259,6 +275,15 @@ class CreateTransactionActivity : AppCompatActivity() {
         super.onSaveInstanceState(outState)
     }
 
+    private fun setFromAddress(addressHex: String) {
+        val address = Address(addressHex)
+        from_address.text = addressHex
+        appDatabase.addressBook.resolveNameAsync(address) {
+            from_address.text = it
+        }
+        currentAddressProvider.setCurrent(address)
```
Comment (ligi) [inline app/src/main/java/org/walleth/activities/CreateTransactionActivity.kt]: think this should only be done when there was a change

**Comment by friedger** [comment]
which balance check? After coming back to the main screen? Are you on riot.im?

**Comment by ligi** [comment]
:+1:

File: app/src/main/java/org/walleth/activities/CreateTransactionActivity.kt [hunk 3]
```diff
@@ -232,6 +208,46 @@ class CreateTransactionActivity : AppCompatActivity() {
...
+        if (to_address.text.isEmpty()) {
+            alert(R.string.create_tx_error_address_must_be_specified)
+        } else if (currentAmount == null) {
+            alert(R.string.create_tx_error_amount_must_be_specified)
+        } else if (currentTokenProvider.currentToken.isETH() && currentAmount!! + gas_price_input.asBigInit() * gas_limit_input.asBigInit() > currentBalanceSafely()) {
+            alert(R.string.create_tx_error_not_enough_funds)
+        } else if (nonce_input.text.isBlank()) {
+            alert(title = R.string.nonce_invalid, message = R.string.please_enter_name)
+        } else {
+            val toAddress = currentAddressProvider.getCurrent()
```
Comment (friedger) [inline app/src/main/java/org/walleth/activities/CreateTransactionActivity.kt]: This is wrong. Sorry. I'll send a fix

**Pull request merged**